### PR TITLE
feat: add animation on scale button

### PR DESCRIPTION
**Pull request by gca-axelor**
RM#76432

### CODE REVIEW

**Comment by gca-axelor** [comment]
storybook down

File: packages/ui/src/components/templates/BottomBar/BarItem.tsx [hunk 1]
```diff
@@ -16,12 +16,16 @@
...
  */
 
-import React, {useMemo} from 'react';
-import {StyleSheet, View} from 'react-native';
+import React, {useEffect, useMemo} from 'react';
+import {StyleSheet, View, Text} from 'react-native';
 import {Color, useThemeColor} from '../../../theme';
 import {checkNullString} from '../../../utils';
 import {Button, NumberBubble} from '../../molecules';
-import {Text} from '../../atoms';
```
Comment (vhu-axelor) [inline packages/ui/src/components/templates/BottomBar/BarItem.tsx:24]: Restore import

File: packages/ui/src/components/templates/BottomBar/BarItem.tsx [hunk 2]
```diff
@@ -56,15 +72,17 @@ const BarItem = ({
...
+          color={buttonColor}
+          disabled={disabled}
+          onPress={onPress}
+          width={size}
+          style={{height: size}}
+          iconSize={size * 0.6}
+        />
+      </Animated.View>
       {indicator > 0 && (
         <NumberBubble
```
Comment (vhu-axelor) [inline packages/ui/src/components/templates/BottomBar/BarItem.tsx:87]: Add animation on `NumberBubble`

File: packages/ui/src/components/templates/BottomBar/BarItem.tsx [hunk 3]
```diff
@@ -56,15 +72,17 @@ const BarItem = ({
...
+        <Button
+          iconName={iconName}
+          color={buttonColor}
+          disabled={disabled}
+          onPress={onPress}
+          width={size}
+          style={{height: size}}
+          iconSize={size * 0.6}
+        />
+      </Animated.View>
```
Comment (vhu-axelor) [inline packages/ui/src/components/templates/BottomBar/BarItem.tsx:85]: When the button size increases, the icon becomes pixelated

File: packages/ui/src/components/templates/BottomBar/BottomBar.tsx [hunk 4]
```diff
@@ -66,6 +105,18 @@ const BottomBar = ({
...
+            styles.animatedBar,
+            animatedStyle,
+            {
+              backgroundColor:
+                selectedItemColor?.background != null
+                  ? selectedItemColor?.background
+                  : Colors.primaryColor?.background,
+            },
+          ]}
+        />
```
Comment (vhu-axelor) [inline packages/ui/src/components/templates/BottomBar/BottomBar.tsx:119]: The bar comes out of the card when moving from one end to the other

**Review by vhu-axelor — CHANGES_REQUESTED** [review]

**Review by vhu-axelor — APPROVED** [review]

**Pull request merged**